Composer/packages/client/src/pages/language-generation/code-editor.tsx

```diff
@@ -59,9 +60,10 @@ const CodeEditor: React.FC<CodeEditorProps> = props => {
   useEffect(() => {
     const currentDiagnostics = inlineMode && template ? filterTemplateDiagnostics(diagnostics, template) : diagnostics;
 
-    const isInvalid = !isValid(currentDiagnostics);
-    const text = isInvalid ? combineMessage(currentDiagnostics) : '';
-    setErrorMsg(text);
+    const errors = findErrors(currentDiagnostics);
+    const warnings = findWarnings(currentDiagnostics);
+    setErrorMsg(combineMessage(errors));
+    setWarningMsg(combineMessage(warnings));
```

Comment:
Not introduced by this PR and we don't need to fix, the problem here is we are setting states inside `useEffect`.
It indicates that either our component props is not well designed and broke the change detection or it could be computed data from props.

Reply:
Correspondingly, I checked the LU editor below and found that these two `code-editor` are sharing a lot of code logic but LU editor is using computed value instead of states, that's confusing.
Seems we just copied one editor to build the other one.

Reply:
The difference is lg save in real time, but lu inline editor only do save when content is valid, that's an issue from luParser may be fixed in future. It decides the error message comes from different place, useState is more flexible here.

Reply:
I got it